lib/Sema/TypeCheckStmt.cpp

```diff
@@ -1143,6 +1153,18 @@ void TypeChecker::checkIgnoredExpr(Expr *E) {
     return;
   }
 
+  bool suggestDiscardResult = false;
+  SWIFT_DEFER {
+    if (suggestDiscardResult) {
+      diagnose(valueE->getLoc(), diag::discard_result_of_call)
+      .fixItInsert(valueE->getStartLoc(), "_ = ");
+    }
```

Comment:
I also considered a fix-it to wrap the call in parentheses, like `({ print("result") }())`. What do you think of that option?

It was a bit tricky to get it to happen at the right time, because starting an expression with a closure only seems to be disallowed when it's part of the *top-level* code (which was a bit surprising to me). But I could probably make it work with some more effort.